### PR TITLE
Use `const auto &` in a few for loops that were not doing it.

### PR DESCRIPTION
**Pull request by ruevs**
Could help with compiler optimizations and thus performance.

### CODE REVIEW

**Review by rpavlik — APPROVED** [review]
Looks good to me as long as it all builds (meaning the called functions are already const-correct) - nice catch! You might also want to get that std::unique predicate in draw.cpp line 71 with the `const&` wand

File: src/draw.cpp [hunk 1]
```diff
@@ -71,7 +71,7 @@ void GraphicsWindow::Selection::Draw(bool isHovered, Canvas *canvas) {
         auto it = std::unique(refs.begin(), refs.end(),
                               [](Vector a, Vector b) { return a.Equals(b); });
```
Comment (rpavlik) [inline src/draw.cpp:72]: unrelated but more low hanging fruit: 

```suggestion
                              [](const Vector &a, const Vector &b) { return a.Equals(b); });
```

Comment (ruevs) [reply src/draw.cpp:72]: Good catch - thank you.

**Comment by ruevs** [comment]
Thanks @rpavlik, added your suggestion and merged.

**Pull request merged**